Enterprise-CMCS/managed-care-review
# Reset deprectatedRateProgramIDs on resubmitted submissions

## Summary

For unlocked submissions 
- On ReviewSubmit no longer display the deprecated "Programs this rate covers field" and display the "You must provide this info" on the new "Rates this rate covers" field if none have been provided
- On editing Rate Details, clear out the `deprecatedRateProgramIDs` field

#### Related issues
https://jiraent.cms.gov/browse/MCR-3804

#### Screenshots

<img width="422" alt="Screenshot 2024-05-28 at 10 37 47 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/b84891ca-a229-417b-9d6a-693e882134af">

<img width="785" alt="Screenshot 2024-05-29 at 10 11 45 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/c13fb0b8-dd46-4906-81f5-325e2081dc88">


#### Test cases covered

- "Programs this rate covers" field doesnt appear on unlocked submissions
- "You must provide missing info" indication is present when unlocked for state users and rate only has historic `deprecatedRateProgramIDs` set

## QA guidance

This PR is testable locally 

- Create and submit a submission with rates
- Add a migration or update code to set the `deprecatedRateProgramsIDs` field. The `rateProgramIDs` field should be `[]`
- As a CMS user unlock the submission
- As a state user on the ReviewSubmit page you should 1) not see the "Programs this rate covers field" and 2) the "Rates this rate covers field" should have an missing info indicator on it
- Click 'Edit' to go to the Rate Details page. Add a program
- On the ReviewSubmit page 1) in the network tab you should see that the `draftRevision` shows that `deprecatedRateProgramIDs` has been cleared out to `[]` and 2) in the UI the programs are listed under "Rates this rate covers" field
